hyper/common/connection.py

```diff
@@ -78,8 +79,10 @@ def __init__(self,
         self._h1_kwargs.update(kwargs)
         self._h2_kwargs.update(kwargs)
 
+        self._timeout = timeout
```

Comment:
This should go into `_h1_kwargs` and `_h2_kwargs`: it'll also need associated tests.